tools/merge_requirements.py

```diff
@@ -53,7 +62,15 @@ def main(*paths):
     """
     data = {}
     for path in paths:
-        data.update(read_file(path))
+        data.update(process_entries(read_file(path)))
+
+    # Need to check if interactive to avoid blocking if nothing is piped
+    if not sys.stdin.isatty():
```

Comment:
I'm a little worried about the mix of file order mattering and taking files from both argv and stdin. What comes first? argv or stdin? We could arbitrarily pick one, but there doesn't seem to be an intuitively correct order to me so I'd personally rather not pick at all.

I think we should either:

1. Stick with just parsing files given on the command line. We actually don't even need this functionality because the files already exist in the Dockerfile. We're just running `cat <file> <file> | merge_requirements` when we can simply run `merge_requirements <file> <file>`. This keeps the code simpler, avoids this problem, and is arguably the better engineering decision, but it makes the UNIX fanboy in me somewhat unhappy...
2. Alternatively, we could only read stdin if no files were provided on the command line. If we take this approach, I think we should make sure to update the docstring of this function/file to explain this behavior.

You're welcome to make the call here between these two approaches (or another one you have in mind?) if you like.

Reply:
We should go with option 2.